src/material.cpp

```diff
@@ -469,13 +497,149 @@ void pyne::Material::write_hdf5(std::string filename, std::string datapath,
+  hid_t fapl;
+  fapl = H5Pcreate(H5P_FILE_ACCESS);
+  H5Pset_fclose_degree(fapl, H5F_CLOSE_STRONG);
+  // Create new/open datafile.
+  hid_t db;
+  if (!pyne::file_exists(filename)) {
+    // Create the file
+    db = H5Fcreate(filename.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, fapl);
+
+  } else {
+    bool ish5 = H5Fis_hdf5(filename.c_str());
+    if (!ish5) throw h5wrap::FileNotHDF5(filename);
+    db = H5Fopen(filename.c_str(), H5F_ACC_RDWR, fapl);
+    
+    // if datapath != /material and exist -> old format
+    if (h5wrap::path_exists(db, datapath) && datapath != "/material") {
+      std::string nucpath;
+      hid_t data_set = H5Dopen2(db, datapath.c_str(), H5P_DEFAULT);
+
+      // Grab the nucpath
+      if (detect_nuclidelist(data_set, nucpath)) {
+        H5Fclose(db);
+        deprecated_write_hdf5(filename, datapath, nucpath, row, chunksize);
+        return;
+      } else {  // can't find a valid nuclide list path form datapath... fail
+        std::cout
+            << "Can't find the nuclide list path in the existing datapath." << std::endl 
+            << "Can't add your material to the datapath."
+            << std::endl;
+        exit(1);
+      }
+    }
+  }
+  // Check if /material exist and what type it is
+  herr_t status;
+  H5O_info_t object_info;
+  status = H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+  status = H5Oget_info_by_name(db, "/material", &object_info, H5P_DEFAULT);
+  // check if "/material" exists
+  if (status == 0) {
+    if (object_info.type ==
+        H5O_TYPE_DATASET) {  // "/material" is a dataset -> old format use
+                             // deprecated write_hdf5
+      H5Fclose(db);
+      deprecated_write_hdf5(filename, datapath, "/nucid", row, chunksize);
+      return;
+
+    } else if (object_info.type != H5O_TYPE_GROUP) {  // "/material" either a dataset or a group:
+                                  // fail!
+      std::cout << "Non-group/non-dataset object /material already exists "
+                   "in the file. Can't write the Material"
+                << std::endl;
+      exit(1);
+    } else {  // "/material" is a group get his hid
+      material_grp_id = H5Gopen2(db, "/material", H5P_DEFAULT);
+    }
+  } else {  // "/material" do not exist -> create it !
+    material_grp_id =
+        H5Gcreate2(db, "/material", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+  }
+  // Check is /material exist as a Group
+  // Group "/material/datapath" does not exist create it
+  if (!h5wrap::path_exists(db, "/material" + datapath)) {
+    data_id = H5Gcreate2(db, ("/material"+datapath).c_str(), H5P_DEFAULT,
+                         H5P_DEFAULT, H5P_DEFAULT);
+  } else {
+    data_id = H5Gopen2(material_grp_id, datapath.c_str(), H5P_DEFAULT);
+  }
+  //std::string nucpath = "/nuclidelist";
+  std::string full_datapath = "/material" + datapath + "/composition";
+  std::string nucpath = "/material" + datapath + "/nuclidelist";
+  std::vector<int> nuclides = write_hdf5_nucpath(data_id, nucpath);
```

Comment:
wanted to use relative path (this is why I stored the different group ids.
But for a reason I ignore it does not work...
So I switched back to absolute path....